### PR TITLE
feat(atomic,headless): add analytics to the show more/show less button

### PR DESCRIPTION
**Pull request by liviarett**
https://coveord.atlassian.net/browse/KIT-1576?atlOrigin=eyJpIjoiZmQxZDZmNjMwOTQwNGUzNWE1NzJmMjM3Y2ZjNTk1MTQiLCJwIjoiaiJ9

### CODE REVIEW

File: packages/headless/src/controllers/folded-result-list/headless-folded-result-list.ts [hunk 1]
```diff
@@ -86,6 +89,16 @@ export interface FoldedResultList extends Controller {
    * @param collection - The collection for which to load more results.
    */
   loadCollection(collection: FoldedCollection): void;
```
Comment (olamothe) [inline packages/headless/src/controllers/folded-result-list/headless-folded-result-list.ts:91]: Hum why not bake in the `dispatch(logShowMoreFoldedResults(result))` inside the `loadCollection` function ?

We want to encourage users as much as possible to log analytics on every interaction, and having it automatically done in the function that you need to call anyway to "show more" seems more foolproof.

Having proper analytics recording is a big pain point for any Coveo deployment, and so we must make sure it's as automatic/transparent as possible for implementers, and that they don't have to go out of their way to remember to call a specific function at the right time.

Comment (liviarett) [reply packages/headless/src/controllers/folded-result-list/headless-folded-result-list.ts:91]: makes sense!

**Review by btaillon — APPROVED** [review]

File: packages/atomic/src/components/result-lists/atomic-result-children/atomic-result-children.tsx [hunk 2]
```diff
@@ -230,7 +237,16 @@ export class AtomicResultChildren {
           part="show-hide-button"
           class="show-hide-button"
           style="text-primary"
-          onClick={() => (this.hideResults = !this.hideResults)}
+          onClick={() => {
+            if (this.hideResults) {
+              this.foldedResultList.logShowMoreFoldedResults(
```
Comment (olamothe) [inline packages/atomic/src/components/result-lists/atomic-result-children/atomic-result-children.tsx:242]: If the event is called when loading a collection, I don't understand why we re-send the event here ?

Doesn't it get sent two times to the API then ?

File: packages/headless/src/controllers/folded-result-list/headless-folded-result-list.ts [hunk 3]
```diff
@@ -172,6 +185,13 @@ export function buildFoldedResultList(
           ] as string
         )
       );
+      dispatch(logShowMoreFoldedResults(collection.result));
+    },
+    logShowMoreFoldedResults: (result) => {
```
Comment (olamothe) [inline packages/headless/src/controllers/folded-result-list/headless-folded-result-list.ts:190]: Related to my above comment, if the event is now dispatched when loading the collection, this function now seems superfluous and could be removed.

File: packages/headless/src/controllers/folded-result-list/headless-folded-result-list.ts [hunk 4]
```diff
@@ -86,6 +89,16 @@ export interface FoldedResultList extends Controller {
...
    */
   loadCollection(collection: FoldedCollection): void;
+  /**
+   * Logs a Click Event for when user loads/shows more folded results.
+   *
+   * @param result - The result for which to send analytics.
+   */
+  logShowMoreFoldedResults(result: Result): void;
+  /**
+   * Logs a Custom Event for when user shows less folded results.
```
Comment (npushkarskii) [inline packages/headless/src/controllers/folded-result-list/headless-folded-result-list.ts:99]:
```suggestion
   * Logs a click event when a user loads/shows more folded results.
   *
   * @param result - The result to send analytics for.
   */
  logShowMoreFoldedResults(result: Result): void;
  /**
   * Logs a custom event when a user shows less folded results.
```

**Review by npushkarskii — APPROVED** [review]
LGTM 👍

**Review by ThibodeauJF — APPROVED** [review]

**Pull request merged**